src/components/header.jsx

```diff
@@ -49,20 +52,72 @@ function Header({ slug }) {
     ),
   );
 
+  const handleClick = (langKey) => {
+    i18n.changeLanguage(langKey);
+    navigate(`/${langKey}/${slug}`);
+  };
+
+  const languageNavItems = (
+    <div className="header__language-nav-container">
+      <div className="header__language-nav-RtoL">
+        {languages.map(
+          (language) => language.isRtoL && (
+          <div className="header__language-nav-items">
+            <Button
+              onClick={() => {
+                handleClick(language.langKey);
+              }}
+              type="button"
+              unstyled
+              key={language.langKey}
+            >
+              {language.lang}
+            </Button>
+          </div>
+          ),
+        )}
+      </div>
+      <div className="header__language-nav-LtoR">
+        {languages.map(
+          (language) => !language.isRtoL && (
+          <div className="header__language-nav-items">
+            <Button
```

Comment:
Repetition still going on from lines 84 -93.

Reply:
Updated to use a function to return language nav items and avoid repetition